app/src/main/java/com/odysee/app/ui/findcontent/AllContentFragment.java

```diff
@@ -466,6 +496,172 @@ public void onPause() {
+                                                request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE);
+
+                                                // get download service and enqueue file
+                                                final DownloadManager manager = (DownloadManager) c.getSystemService(Context.DOWNLOAD_SERVICE);
+                                                manager.enqueue(request);
+
+                                                //set BroadcastReceiver to install app when .apk is downloaded
+                                                BroadcastReceiver onComplete = new BroadcastReceiver() {
+                                                    public void onReceive(Context ctxt, Intent intent) {
+                                                        ContextCompat.checkSelfPermission(ctxt, Manifest.permission.READ_EXTERNAL_STORAGE);
+
+                                                        installApk();
+                                                        getContext().unregisterReceiver(this);
+                                                    }
+                                                };
+                                                //register receiver for when .apk download is compete
+                                                c.registerReceiver(onComplete, new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
+                                            }
+                                        }
+                                    }).show();
+                                }
+                            });
+                        }
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+
+        });
+    }
+
+    @SuppressLint("QueryPermissionsNeeded")
+    private void installApk() {
+        try {
+            Context context = getContext();
+            if (context != null) {
+                File file = new File(getContext().getExternalFilesDir(null).getAbsolutePath().concat("/odysee-android.apk"));
+                Intent intent = new Intent(Intent.ACTION_VIEW);
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
+                    Uri downloaded_apk = FileProvider.getUriForFile(context, context.getApplicationContext().getPackageName() + ".provider", file);
+                    intent.setDataAndType(downloaded_apk, "application/vnd.android.package-archive");
+                    List<ResolveInfo> resInfoList;
+
+                    if (Build.VERSION.SDK_INT > Build.VERSION_CODES.S_V2) {
+                        resInfoList = context.getPackageManager().queryIntentActivities(intent, PackageManager.ResolveInfoFlags.of(PackageManager.MATCH_DEFAULT_ONLY));
+                    } else {
+                        //noinspection deprecation
```

Comment:
Android Studio says this is a "redundant suppression", and the fixit (when pressing alt+enter) removes this line.

Reply:
Same as above